tests/admin.test.ts

```diff
@@ -182,3 +185,180 @@ describe('bonus points submission', () => {
+
+
+    users[0].accessType = UserAccessType.STAFF;
+    users[1].accessType = UserAccessType.STANDARD;
+    users[2].accessType = UserAccessType.MARKETING;
+    users[3].accessType = UserAccessType.MERCH_STORE_DISTRIBUTOR;
+
+    await new PortalState()
+      .createUsers(users[0], users[1], users[2], users[3], admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    const accessLevelResponse = await adminController.updateUserAccessLevel({
+      "accessUpdates": [
+        { "user": users[0].email,  "newAccess": {
+            "accessType": "MERCH_STORE_MANAGER"
+        }},
+        { "user": users[1].email,  "newAccess": {
+            "accessType": "MARKETING"
+        }},
+        { "user": users[2].email,  "newAccess": {
+            "accessType": "MERCH_STORE_DISTRIBUTOR"
+        }},
+        { "user": users[3].email,  "newAccess": {
+            "accessType": "STAFF"
+        }}
+      ]
+    }, admin);
+
+    // expect statements
+    expect(accessLevelResponse.updatedUsers[0].email).toEqual(users[0].email);
+    expect(accessLevelResponse.updatedUsers[0].accessType).toEqual(UserAccessType.MERCH_STORE_MANAGER);
+
+    expect(accessLevelResponse.updatedUsers[1].email).toEqual(users[1].email);
+    expect(accessLevelResponse.updatedUsers[1].accessType).toEqual(UserAccessType.MARKETING);
+
+    expect(accessLevelResponse.updatedUsers[2].email).toEqual(users[2].email);
+    expect(accessLevelResponse.updatedUsers[2].accessType).toEqual(UserAccessType.MERCH_STORE_DISTRIBUTOR);
+
+    expect(accessLevelResponse.updatedUsers[3].email).toEqual(users[3].email);
+    expect(accessLevelResponse.updatedUsers[3].accessType).toEqual(UserAccessType.STAFF);
+  });
+
+
+
+  // attempt to update when user is not an admin
+  test('attempt to update when user is not an admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const standard = UserFactory.fake({ accessType: UserAccessType.STANDARD });
+    const users = UserFactory.create(5);
+
+
+    users[0].accessType = UserAccessType.STAFF;
+    users[1].accessType = UserAccessType.STANDARD;
+    users[2].accessType = UserAccessType.MARKETING;
+    users[3].accessType = UserAccessType.MERCH_STORE_DISTRIBUTOR;
+
+    await new PortalState()
+      .createUsers(users[0], users[1], users[2], users[3], standard)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        "accessUpdates": [
+          { "user": users[0].email, "newAccess": { "accessType": "MERCH_STORE_MANAGER" } },
+          { "user": users[1].email, "newAccess": { "accessType": "MARKETING" } },
+          { "user": users[2].email, "newAccess": { "accessType": "MERCH_STORE_DISTRIBUTOR" } },
+          { "user": users[3].email, "newAccess": { "accessType": "STAFF" } }
+        ]
+      }, standard);
+    }).rejects.toThrow(ForbiddenError);
+  });
```

Comment:
Let's also add another test making sure the role of the user is not updated

Reply:
the role of all users in the input of the changes requested

Reply:
the goal is that when an error is thrown, the request should not make changes to the database

Reply:
let me know if there is any questions regarding for what I requested